cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -692,6 +694,54 @@ io.cdap.cdap.runtime.spi.provisioner.ClusterStatus getClusterStatus(String name)
     return status;
   }
 
+  String getClusterFailureMsg(String name) throws RetryableProvisionException {
+    io.cdap.cdap.runtime.spi.provisioner.ClusterStatus status = getDataprocCluster(name)
+      .map(tempCluster -> convertStatus(tempCluster.getStatus()))
+      .orElse(io.cdap.cdap.runtime.spi.provisioner.ClusterStatus.NOT_EXISTS);
+
+    // if it failed, try to get the create operation and log the error message
+    if (status == io.cdap.cdap.runtime.spi.provisioner.ClusterStatus.FAILED) {
+      String resourceName = String.format("projects/%s/regions/%s/operations", conf.getProjectId(), conf.getRegion());
+      String filter = String.format("clusterName=%s AND operationType=CREATE", name);
```

Comment:
I guess you are suggesting we add operationId and status to Cluster? If we add new fields to Cluster it will break backward compatibility, so should we add as properties?

I am not sure we can skip the status check. Should getClusterStatus which will probably be called before getClusterFailureMsg update the status in Cluster? But cluster object is immutable. I created a JIRA to track https://cdap.atlassian.net/browse/CDAP-19641.

Reply:
Discussed with Albert offline. We decided to keep the method. getClusterFailureMsg will not check for status. And getClusterStatus will not get the failure msg and do a redundant LOG.warn.